src/zc/buildout/easy_install.py

```diff
@@ -399,11 +400,11 @@ def _call_easy_install(self, spec, ws, dest, dist):
             for d in dists:
                 newloc = os.path.join(dest, os.path.basename(d.location))
                 if os.path.exists(newloc):
-                    if os.path.isdir(newloc):
-                        shutil.rmtree(newloc)
-                    else:
-                        os.remove(newloc)
-                os.rename(d.location, newloc)
+                    logger.warn(
+                        "Path %s already exists.\nIf this contains a wrong "
+                        "package, please remove it yourself.", newloc)
+                else:
+                    os.rename(d.location, newloc)
```

Comment:
While the change of another buildout doing its own the `os.rename()` between this buildout calling `exists()` and then `rename()` is slim, it could still happen.

I suggest instead just doing the `os.rename()` first, then if it fails, check that the destination exists (and perhaps try creating a `pkg_resources.Environment([newloc])` on it and checking it), and then give up with a warning. Else raise an error.